common/flatpak-utils.c

```diff
@@ -3207,6 +3207,141 @@ flatpak_mtree_create_root (OstreeRepo        *repo,
   return TRUE;
 }
 
+static gboolean
+validate_mode (guint32 mode,
+               GError **error)
+{
+  if (mode & S_ISUID)
+    return glnx_throw (error, "suid");
+  if (mode & S_ISGID)
+    return glnx_throw (error, "sgid");
+  if (mode & S_IWOTH)
+    return glnx_throw (error, "world-writable");
```

Comment:
I like the non-world-writable check, but that really has to be checked in existing flatpak repos to make sure we don't have anything like that live atm.